web/src/components/betaBar/BetaBar.test.tsx

```diff
@@ -0,0 +1,178 @@
+      expect(screen.getByTestId("feedback-link").getAttribute("href")).toEqual(
+        Config.betaBar.betaFormLinkStarting
+      );
+    });
+  });
+
+  describe("request a feature", () => {
+    it("does not show feature request button when userData is undefined", () => {
+      useUndefinedUserData();
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.queryByText(Config.betaBar.betaModalFeatureRequestButtonText)).not.toBeInTheDocument();
+    });
+
+    it("clears feature request input when modal is closed and is revisted", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByLabelText("close"));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("");
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
+
+    it("submitting feature request renders submission successful modal", async () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByText(Config.betaBar.featureRequestModalSubmitButtonText));
+
+      await waitFor(() => {
+        expect(screen.getByText(Config.betaBar.successfulSubmissionModalButtonText)).toBeInTheDocument();
+      });
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
+
+    it("does not submit feature request when input is empty", () => {
```

Comment:
this name could be more specific, it's not testing JUST that it does not submit but also that it shows an error